crates/cxx-qt-gen/src/parser/mod.rs

```diff
@@ -25,9 +29,67 @@ use syn::{
     punctuated::Punctuated,
     spanned::Spanned,
     token::{Brace, Semi},
-    Error, Ident, Item, ItemMod, Meta, Result, Token,
+    Attribute, Error, ForeignItemFn, Ident, Item, ItemMod, Meta, Result, Token,
 };
 
+fn check_safety(method: &ForeignItemFn, safety: &Safety) -> Result<()> {
+    if safety == &Safety::Unsafe && method.sig.unsafety.is_none() {
+        Err(Error::new(
+            method.span(),
+            "Invokables must be marked as unsafe or wrapped in an `unsafe extern \"RustQt\"` block!",
+        ))
+    } else {
+        Ok(())
+    }
+}
+
+pub trait Invokable {
+    fn name(&self) -> &Name;
+}
+
+/// Struct with common fields between Invokable types.
+/// These types are ParsedSignal, ParsedMethod and ParsedInheritedMethod
+pub struct MethodFields {
+    qobject_ident: Ident,
+    mutable: bool,
+    parameters: Vec<ParsedFunctionParameter>,
+    safe: bool,
+    name: Name,
+    docs: Vec<Attribute>, // TODO: Remove this
+}
+
+/// Function for creating an [MethodFields] from a method and docs.
+/// These fields are shared by ParsedSignal, ParsedMethod and ParsedInheritedMethod so grouped into common logic.
+pub fn extract_common_fields(method: &ForeignItemFn, docs: Vec<Attribute>) -> Result<MethodFields> {
```

Comment:
Why isn't this function `MethodFields::parse`?
That's the pattern we use for all other parsing.

Reply:
Done